aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -371,6 +371,37 @@ static void polygamma_kernel(TensorIterator& iter, int64_t n) {
   }
 }
 
+static void nan_to_num_kernel(
+    TensorIterator& iter,
+    c10::optional<double> nan,
+    c10::optional<double> pos_inf,
+    c10::optional<double> neg_inf) {
+  if (c10::isIntegralType(iter.dtype(), true)) {
```

Comment:
Should nan_to_num just call clone on non-float, non-complex types?

Reply:
I was thinking about it. But since `out` is a kwarg, for correctly handling and using the passed tensor, I felt using the iterator and identity operator seemed better.

Reply:
Hm... wouldn't it just be this check with:

```
result.copy_(self);
```

? 

That seems pretty straightforward.

Reply:
(you might have to resize result using `resize_output()` first)